src/main/scala/org/tygus/suslik/language/Expressions.scala

```diff
@@ -31,6 +40,16 @@ object Expressions {
   trait SymmetricOp
   trait AssociativeOp
 
+  object OpOverloadedEq extends  OverloadedBinOp{
+    override def level: Int = 3
+    override def pp: String = "=="
+    override def opFromTypes: Map[(SSLType, SSLType), BinOp] = Map(
+      (IntType, IntType) -> OpEq,
+      (IntSetType, IntSetType) -> OpSetEq,
+      (BoolType, BoolType) -> OpEq,
```

Comment:
Does opEq work correctly on Booleans? I thought we had to use a bi-implication to encode it. Can you double-check that it works? (Add some tests)

Reply:
I can't really check it yet, because it ends up in a formula that is not in CNF in all scenarios that can test Boolean equality behavior.